bot/exts/info/information.py

```diff
@@ -533,21 +553,43 @@ async def rules(self, ctx: Context, rules: Greedy[int]) -> None:
             await ctx.send(embed=rules_embed)
             return
 
-        full_rules = await self.bot.api_client.get("rules", params={"link_format": "md"})
-
         # Remove duplicates and sort the rule indices
-        rules = sorted(set(rules))
+        rule_numbers = sorted(set(rule_numbers))
+        # Remove duplicate keywords and preserve the order of initial keywords
+        keywords = list(dict.fromkeys(keywords))
 
-        invalid = ", ".join(str(index) for index in rules if index < 1 or index > len(full_rules))
+        invalid = ", ".join(
+            str(rule_number) for rule_number in rule_numbers
+            if rule_number < 1 or rule_number > len(full_rules))
 
         if invalid:
             await ctx.send(shorten(":x: Invalid rule indices: " + invalid, 75, placeholder=" ..."))
             return
 
-        for rule in rules:
-            self.bot.stats.incr(f"rule_uses.{rule}")
+        final_rule_numbers = []
+        final_rule_numbers.extend(rule_numbers)
+
+        for keyword in keywords:
+            for pick, rule in enumerate(full_rules):
+                if keyword in rule[1]:
+                    final_rule_numbers.append(pick + 1)
+                    break
+
+        if not rule_numbers and not final_rule_numbers:
+            # This would mean that only keywords where used and no match for them was found
+            await ctx.send(
+                f"There are currently no rules that correspond to keywords: **{', '.join(keywords)}**.\n"
+                f"If you think it should be added, please suggest it in either "
+                f"<#{constants.Channels.meta}> or <#{constants.Channels.dev_contrib}>")
+            return
+
+        final_rules = []
+        final_rule_numbers.sort()
+        final_rule_numbers = set(final_rule_numbers)
```

Comment:
sets are not ordered, so you should remove duplicates and *then* sort.

(in practice they will most likely end up ordered anyway because of how integers are hashed, although this isn't something to rely on)